### PR TITLE
Using miniredis instead of real redis container for unit tests

### PR DESCRIPTION
**Pull request by apourchet**

### CODE REVIEW

File: tracker/peerstore/redis.go [hunk 1]
```diff
@@ -19,11 +19,11 @@ import (
 	"strconv"
 	"strings"
 
+	"github.com/andres-erbsen/clock"
+	"github.com/garyburd/redigo/redis"
```
Comment (yiranwang52) [inline tracker/peerstore/redis.go]: Can you add a space and put external dependencies below internal dependencies?

File: tracker/peerstore/redis.go [hunk 2]
```diff
@@ -170,8 +170,12 @@ func (s *RedisStore) GetPeers(h core.InfoHash, n int) ([]*core.PeerInfo, error)
 
 	var i int
 	for len(selected) < n && i < len(windows) {
-		result, err := redis.Strings(c.Do("SRANDMEMBER", peerSetKey(h, windows[i]), n-len(selected)))
-		if err != nil {
+		k := peerSetKey(h, windows[i])
+		result, err := redis.Strings(c.Do("SRANDMEMBER", k, n-len(selected)))
+		if err == redis.ErrNil {
+			i++
```
Comment (yiranwang52) [inline tracker/peerstore/redis.go]: can you move i++ into the `for` clause?

File: tracker/peerstore/redis_test.go [hunk 3]
```diff
@@ -14,31 +14,32 @@
...
 )
 
-func redisConfigFixture() RedisConfig {
-	return RedisConfig{
-		Addr:              "localhost:6380",
-		PeerSetWindowSize: 30 * time.Second,
-		MaxPeerSetWindows: 4,
-	}
+func TestMain(m *testing.M) {
+	rand.Seed(0)
```
Comment (codygibb) [inline tracker/peerstore/redis_test.go]: Why is this needed?

Comment (apourchet) [reply tracker/peerstore/redis_test.go]: Thought we would need this for miniredis to have reproducible random numbers but we don't care about that in the tests.
Will remove.

File: tracker/peerstore/redis.go (hunk 2 above)
Comment (apourchet) [reply tracker/peerstore/redis.go]: Done;

File: tracker/peerstore/redis.go (hunk 1 above)
Comment (apourchet) [reply tracker/peerstore/redis.go]: Done.

File: tracker/peerstore/redis.go [hunk 4]
```diff
@@ -168,10 +169,13 @@ func (s *RedisStore) GetPeers(h core.InfoHash, n int) ([]*core.PeerInfo, error)
...
 
-	var i int
-	for len(selected) < n && i < len(windows) {
-		result, err := redis.Strings(c.Do("SRANDMEMBER", peerSetKey(h, windows[i]), n-len(selected)))
-		if err != nil {
+	for i := 0; len(selected) < n && i < len(windows); i++ {
+		k := peerSetKey(h, windows[i])
+		result, err := redis.Strings(c.Do("SRANDMEMBER", k, n-len(selected)))
+		if err == redis.ErrNil {
+			i++
```
Comment (yiranwang52) [inline tracker/peerstore/redis.go]: Then you can remove this i++ and the one at the end of the loop.
